### PR TITLE
Project Red

### PR DESCRIPTION
**Pull request by bdistin**
### Description of the PR
Full TypeScript rewrite to utilize [@klasa/core](https://github.com/dirigeants/core)

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- Too many to list (feel free to change this). Rewrite everything to use @klasa/core

### Semver Classification

- [ ] This PR only includes documentation or non-code changes.
- [ ] This PR fixes a bug and does not change the (intended) framework interface.
- [ ] This PR adds methods or properties to the framework interface.
- [x] This PR removes or renames methods or properties in the framework interface.

### CODE REVIEW

File: guides/Piece Basics/CreatingExtendables.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Extendables are functions that extend current Discord.js classes by adding methods or properties.
+Extendables are pieces that extend objects or classes by copying its properties and methods like a blueprint.
```
Comment (cfanoulis) [inline guides/Piece Basics/CreatingExtendables.md:1]: hmm, this doesn't really make sense
```suggestion
Extendables are pieces that extend objects or classes by adding, editing or removing properties and methods.
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        rsync -vau dist/src out/dist
+        echo -e "\n# Commit and push"
+        cd out
+        git add --all .
+        git config user.name "${GITHUB_ACTOR}"
+        git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+        git commit -m "TSC build: ${GITHUB_SHA}" || true
+        git push origin $TARGET_BRANCH
+      env:
+        GITHUB_TOKEN: ${{ secrets.COMMIT_TOKEN }}
```
Comment (cfanoulis) [inline .github/workflows/build.yml]: nit: End of line please

File: src/arguments/boolean.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Argument, ArgumentStore, KlasaMessage, Possible } from 'klasa';
+const truths = ['1', 'true', '+', 't', 'yes', 'y'];
+const falses = ['0', 'false', '-', 'f', 'no', 'n'];
```
Comment (cfanoulis) [inline src/arguments/boolean.ts:3]: maybe those 2 should be exported or added as static properties to the Argument

File: src/arguments/channel.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Argument, Possible, KlasaMessage } from 'klasa';
+import { Channel } from '@klasa/core';
+
+export default class CoreArgument extends Argument {
+
+	public async run(argument: string, possible: Possible, message: KlasaMessage): Promise<Channel> {
+		// Regular Channel support
+		const channelID = Argument.regex.channel.exec(argument);
+		const channel = channelID ? await this.client.channels.fetch(channelID[1]).catch(() => null) : null;
```
Comment (cfanoulis) [inline src/arguments/channel.ts:9]: can't we use those new fancy nullish operators here

File: src/arguments/date.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Argument, KlasaMessage, Possible } from 'klasa';
+
+export default class CoreArgument extends Argument {
+
+	public run(argument: string, possible: Possible, message: KlasaMessage): Date {
+		const date = new Date(argument);
+		if (!isNaN(date.getTime()) && date.getTime() > Date.now()) return date;
```
Comment (cfanoulis) [inline src/arguments/date.ts:7]: why not be able to have dates *before* the current time?

File: src/arguments/channel.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+		if (channel) return channel;
+
+		// DM Channel support
+		const userID = Argument.regex.userOrMember.exec(argument);
```
Comment (cfanoulis) [inline src/arguments/channel.ts:13]: why not run the dmChannel argument here

File: guides/Getting Started/GettingStarted.md [hunk 7]
```diff
@@ -5,7 +5,7 @@
 You can install this branch of klasa with the following command:
 
 ```sh
-npm install --save discordjs/discord.js dirigeants/klasa#{@branch}
+npm install --save @klasa/core dirigeants/klasa#{@branch}
```
[the remark this answers is not in the document]
Comment (CyberiumShadow) [reply guides/Getting Started/GettingStarted.md:8]: It’s currently set to the PR branch so we don’t need to keep releasing Klasa core whilst in dev

File: src/arguments/channel.ts (hunk 4 above)
Comment (kyranet) [reply src/arguments/channel.ts:9]: No.

Comment (cfanoulis) [reply src/arguments/channel.ts:9]: ok. /resolved

File: src/arguments/boolean.ts (hunk 3 above)
Comment (bdistin) [reply src/arguments/boolean.ts:3]: Not changing existing behavior at this time. You can still transfer and edit this piece.

File: src/arguments/channel.ts (hunk 6 above)
Comment (bdistin) [reply src/arguments/channel.ts:13]: extra overhead

File: src/arguments/date.ts (hunk 5 above)
Comment (bdistin) [reply src/arguments/date.ts:7]: Not changing the existing behavior at this time. This piece can still be transferred and edited.

File: guides/Getting Started/GettingStarted.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (cfanoulis) [reply guides/Getting Started/GettingStarted.md:8]: ΟΚ. /resolved

**Pull request merged**